server/src/test/java/org/elasticsearch/search/aggregations/support/ValuesSourceRegistryTests.java

```diff
@@ -56,4 +58,61 @@ public void testAggregatorNotFoundException() {
         expectThrows(IllegalArgumentException.class, () -> registry.getAggregator(key, fieldOnly));
         expectThrows(IllegalArgumentException.class, () -> registry.getAggregator(key, scriptOnly));
     }
+
+    public void testValueSourceTypeNullUnregisteredExceptionHandler() {
```

Comment:
I feel weird saying this, but I think this test is unnecessary to the point of being distracting.  You are literally just testing that the assert works, which isn't going to run in production anyway.  So this test doesn't describe the production behavior, which is a little troubling to me.  I think the assert by itself is enough.

Reply:
I agree it is pretty much useless by itself, but I wanted to have a test just to stress the point that a non-null exception is required. The other option was to have some Javadoc...but if I have to choose between documentation and a test I usually prefer a test for two reasons:
1. a test fails if something is wrong
2. a test is also a piece of documentation
3. it is very easy to forget updating documentation while you will have a test failing and you have to fix it if something changes

Reply:
Anyway I can get rid of it.

Reply:
Point 2 is exactly my concern - to me, this test documents that if you return a null exception, we still throw, which is what we want.  That seems to say "it's okay to have a null here".

Since our tests run with asserts enabled, a test will fail if we return a null from `getUnregisteredException()`; that's the whole point of the assert being there, and that's the behavior we want to enforce.

Reply:
Yes right...but if the assert is removed as a result of a code change than something unpredictable can happen. While,  if we have a test checking for the assertion error we would, at least have, a test failing because of the missing assertion. Am I wrong?

Reply:
Anyway if the assert is removed for whatever reason and `unmappedException` is null then we have:

```
Unexpected exception type, expected AssertionError but got java.lang.NullPointerException: Cannot throw exception because "unmappedException" is null
```

Reply:
Of course when a production build is created the assertion goes away but at least the CI would have the test checking that the assertion is still there.